### PR TITLE
chore: cherry-pick d4564dcc2520 from chromium

### PR DESCRIPTION
**Pull request by nornagon**
Reland: Sequentialise access to callbacks in DWriteFontLookupTableBuilder

This reland fixes the sequenced task runner initialisation. Set the
task runner in the constructor and reset it for each unit test execution.

Since there may be multiple instance of DWriteFontProxyImpl instantiated
for multiple RenderProcessHosts, and
DWriteFontProxyImpl::GetUniqueNameLookupTable may access
DWriteFontLookupTableBuilder::QueueShareMemoryRegionWhenReady from
separate threads, there may be race conditions around the
pending_callbacks_ member of DWriteFontLookupTableBuilder.

Sequentialise and guard access to pending_callbacks_ with a separate
sequenced task runner.

Remove explicit configuration of the FontUniqueNameBrowserTest cache
directory as [1] introduced a callback function in
ShellContentBrowserClient which sets this correctly. This avoids
instantiating DWriteFontLookupTableBuilder too early when the ThreadPool
is not yet available in a BrowserTest.

[1] https://chromium-review.googlesource.com/c/chromium/src/+/1776358/9/content/shell/browser/shell_content_browser_client.cc#422

Fixed: 1047054
Change-Id: I38cf8b84a48315980624b68bbf55d3727be457b8
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/2032119
Reviewed-by: Matthew Denton <mpdenton@chromium.org>
Commit-Queue: Dominik Röttsches <drott@chromium.org>
Cr-Commit-Position: refs/heads/master@{#737466}

Notes: Security: Backport fix for buffer underflow in DWrite.

### CODE REVIEW

File: patches/chromium/reland_sequentialise_access_to_callbacks_in.patch [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+   }
+diff --git a/content/browser/renderer_host/dwrite_font_lookup_table_builder_win.cc b/content/browser/renderer_host/dwrite_font_lookup_table_builder_win.cc
+index f8ba10769691864f59ea6f2ef55700d0cd9c4969..c5beb1fb980c0b42e00508e09179c83dfe48e9cf 100644
+--- a/content/browser/renderer_host/dwrite_font_lookup_table_builder_win.cc
++++ b/content/browser/renderer_host/dwrite_font_lookup_table_builder_win.cc
+@@ -23,6 +23,7 @@
+ #include "base/strings/utf_string_conversions.h"
+ #include "base/task/post_task.h"
+ #include "base/task/task_traits.h"
++#include "base/task/thread_pool.h"
```
Comment (deepak1556) [inline patches/chromium/reland_sequentialise_access_to_callbacks_in.patch:72]: `thread_pool` api is not available in this version. `base::ThreadPool::CreateSequencedTaskRunner`

Comment (nornagon) [reply patches/chromium/reland_sequentialise_access_to_callbacks_in.patch:72]: rats.

File: patches/chromium/.patches [hunk 2]
```diff
@@ -100,4 +100,6 @@ move_readablestream_requests_onto_the_stack_before_iteration.patch
 streams_convert_state_dchecks_to_checks.patch
 audiocontext_haspendingactivity_unless_it_s_closed.patch
 protect_automatic_pull_handlers_with_mutex.patch
+reland_sequentialise_access_to_callbacks_in.patch
+cherry-pick-adc8f05aa3ab.patch
```
Comment (deepak1556) [inline patches/chromium/.patches:104]: stale :)

**Comment by nornagon** [comment]
CI failures unrelated; merging.

**Pull request merged**